doc/dev/kernel.xml

```diff
@@ -374,60 +375,105 @@ which does exactly this.
 <Section Label="Sect-Interfaces">
 <Heading>Interfaces</Heading>
 
-The modules at the bottom of the picture from the section 
+The modules at the bottom of the picture from the section
 <Ref Label="Sect-BigPicture" /> (<E>objects</E>, <E>gasman</E> and
-<E>system</E>), and white boxes (<E>arithmetic</E>, <E>lists</E>, 
-<E>calls/operations</E>, <E>records</E>) 
+<E>system</E>), and white boxes (<E>arithmetic</E>, <E>lists</E>,
+<E>calls/operations</E>, <E>records</E>)
 provide interfaces intended for use in the kernel.
 <P/>
 
-For example <F>ariths.h</F> provides macros like 
-<C>SUM</C>, <C>PROD</C>, <C>AINV</C>, 
+For example <F>ariths.h</F> provides functions like
+<C>SUM</C>, <C>PROD</C>, <C>AINV</C>,
 corresponding to the GAP operations <C>+</C>, <C>*</C> and unary <C>-</C>.
-These macros can be applied to any values (objects) to 
-perform an appropriate action. Note that there is some overhead in 
-using these general macros, if you know what your arguments are, 
+These functions can be applied to any values (objects) to
+perform an appropriate action. Note that there is some overhead in
+using these general functions, if you know what your arguments are,
 there may be faster ways.
 <P/>
 
 Another interface provides <C>ELM_LIST</C>, <C>ASS_LIST</C>,
 <C>LEN_LIST</C>, etc. as a general interface to any type of list.
 <P/>
 
-Macros like these will even work for &GAP;-level objects whose 
+Functions like these will even work for &GAP;-level objects whose
 arithmetic or list operations are implemented by installed methods.
 <P/>
 
-Adding a kernel function is easy if it 
-stays in the region within the thick lines, 
-i.e. it uses interfaces from white and yellow areas, 
+Adding a kernel function is easy if it
+stays in the region within the thick lines,
+i.e. it uses interfaces from white and yellow areas,
 which provide kernel equivalents to the basic GAP functionality.
 <P/>
 
-For example, the following kernel function will return a list containing an 
-object, its square and its cube: 
+For example, the following kernel function will return a list containing an
+object, its square and its cube:
 
 <Log>
 <![CDATA[
 /*  x -> [x,x^2,x^3]  */
- Obj FuncFoo1(Obj self, Obj x) {
- Obj x2, x3;
- Obj l = NEW_PLIST( T_PLIST, 3 );
- SET_ELM_PLIST( l, 1, x );
- CHANGED_BAG( l );
- x2 = PROD( x, x );
- SET_ELM_PLIST( l, 2, x2 );
- CHANGED_BAG( l );
- x3 = PROD( x2, x );
- SET_ELM_PLIST( l, 3, x3 );
- CHANGED_BAG( l );
- SET_LEN_PLIST(l, 3);
- CHANGED_BAG( l );
- return l;
- }
+Obj FuncFoo1(Obj self, Obj x)
+{
+    Obj x2, x3;
+    Obj l = NEW_PLIST( T_PLIST, 3 );
```

Comment:
I hereby remind you of your own suggestion:
```suggestion
    Obj list = NEW_PLIST( T_PLIST, 3 );
```
And for the rest as well.

Reply:
Thanks :)